var/spack/repos/builtin/packages/mozjs/package.py

```diff
@@ -31,15 +31,24 @@ class Mozjs(AutotoolsPackage):
     configure_directory = 'js/src'
     build_directory = 'js/src/spack-build'
 
-    patch('perl-bug.patch')
+    patch('perl-bug.patch', sha256='9f7d8502d85a4125e975a84cae11a8b34e696172d56f8ebc7ecf6d21fa3c30c9')
+    patch('Bug-638056-Avoid-The-cacheFlush-support-is-missing-o.patch',
+          sha256='b1c869a65f5ebc10741d4631cc2e1e166c6ed53035cfa56bede55a4c19b7b118', when='@1.8.5')
+    patch('fix-811665.patch',
+          sha256='2b298b8a693865b38e2b0d33277bb5ffe152c6ecf43648e85113fec586aa4752', when='@1.8.5')
 
     def configure_args(self):
         spec = self.spec
-        return [
+        config_args = [
             '--enable-readline',    # enables readline support in JS shell
             '--enable-threadsafe',  # enables support for multiple threads
             '--enable-system-ffi',
             '--with-system-zlib={0}'.format(spec['zlib'].prefix),
             '--with-system-nspr',
             '--with-nspr-prefix={0}'.format(spec['nspr'].prefix),
         ]
+        if spec.target.family == 'aarch64':
+            config_args.append('--host=aarch64-linux-gnu')
+        if spec.satisfies('@1.8.5'):
+            config_args.append('--disable-readline')
```

Comment:
I think it's also reasonable to update the `depends_on` statement for `readline` like: `depends_on('readline', when='@17.0.0:')`

Reply:
fixed in https://github.com/spack/spack/pull/17594/commits/24e62fd4ca57b695cf80754394454d0e173dcae9